docs/reference/transfers.md

```diff
@@ -3,14 +3,150 @@
 A `transfer` is an immutable record of a financial transaction between
 two accounts.
 
+In TigerBeetle, financial transactions are called "transfers" instead of "transactions" because
+the latter term is heavily overloaded in the context of databases.
+
+### Updates
+
+Transfers *cannot be modified* after creation.
+
+
+## Modes
+
+Fields used by each mode of transfer:
+
+| Field                         | Single-Phase | Pending  | Post-Pending | Void-Pending |
+| ----------------------------- | ------------ | -------- | ------------ | ------------ |
+| `id`                          | required     | required | required     | required     |
+| `debit_account_id`            | required     | required | optional     | optional     |
+| `credit_account_id`           | required     | required | optional     | optional     |
+| `user_data`                   | optional     | optional | optional     | optional     |
+| `reserved`                    | none         | none     | none         | none         |
+| `pending_id`                  | none         | none     | required     | required     |
+| `timeout`                     | none         | required | none         | none         |
+| `ledger`                      | required     | required | optional     | optional     |
+| `code`                        | required     | required | optional     | optional     |
+| `flags.linked`                | optional     | optional | optional     | optional     |
+| `flags.pending`               | false        | true     | false        | false        |
+| `flags.post_pending_transfer` | false        | false    | true         | false        |
+| `flags.void_pending_transfer` | false        | false    | false        | true         |
+| `amount`                      | required     | required | optional     | optional     |
+
 TigerBeetle uses the same data structures internally and
 externally. This means that sometimes you need to set temporary values
 for fields that TigerBeetle, not you (the user), are responsible.
```

Comment:
This section has kind of gotten lost now. I'd suggest moving it to above the updates section like it was before or removing it